services/metadata-utils/src/main/java/datawave/query/util/AllFieldMetadataHelper.java

```diff
@@ -52,18 +53,24 @@
 import java.util.Set;
 import java.util.concurrent.ExecutionException;
 
-/**
- */
-@Configuration
 @EnableCaching
 @Component("allFieldMetadataHelper")
+@Scope("prototype")
 public class AllFieldMetadataHelper {
-    private static final Logger log = Logger.getLogger(AllFieldMetadataHelper.class);
+    private static final Logger log = LoggerFactory.getLogger(AllFieldMetadataHelper.class);
     
     public static final String NULL_BYTE = "\0";
     
     protected static final Function<MetadataEntry,String> toFieldName = new MetadataEntryToFieldName(), toDatatype = new MetadataEntryToDatatype();
     
+    public AllFieldMetadataHelper(TypeMetadataHelper typeMetadataHelper, CompositeMetadataHelper compositeMetadataHelper) {
+        Preconditions.checkNotNull(typeMetadataHelper);
+        this.typeMetadataHelper = typeMetadataHelper;
+        
+        Preconditions.checkNotNull(compositeMetadataHelper);
+        this.compositeMetadataHelper = compositeMetadataHelper;
+    }
+    
     protected String getDatatype(Key k) {
         String datatype = k.getColumnQualifier().toString();
         int index = datatype.indexOf('\0');
```

Comment:
Not related to your changes, but I notice we define a NULL_BYTE constant, yet we don't use it here

Reply:
> Not related to your changes, but I notice we define a NULL_BYTE constant, yet we don't use it here

The NULL_BYTE constant is a string, whereas we're searching for a char here. I'm not sure if there's a performance impact to doing indexOf on a single character string vs a character, but that is one difference.